FlowCrypt/Functionality/Services/Compose Message Service/ComposeMessageService.swift

```diff
@@ -62,33 +64,41 @@ final class ComposeMessageService {
         includeAttachments: Bool = true,
         signingPrv: PrvKeyInfo?
     ) async throws -> SendableMsg {
+        state = .validatingMessage
+
         let recipients = contextToSend.recipients
         guard recipients.isNotEmpty else {
+            state = .error
```

Comment:
error state should be set once at the error handler, not each time where error is thrown

Reply:
This is indicates state for service, not for clients code.

Reply:
Will get rid of it with "callbacks" rework

Reply:
It does now, but it doesn't have to be implemented that way. It doesn't have to be implemented as a state, instead a callback can be passed around to update progress element.

In fact, since errors are all handled with a modal, there doesn't need to be any progress element update when errors happen.